docs/rst/knora-api-server/api_v1/adding-resources.rst

```diff
@@ -132,70 +126,232 @@ The resource's Iri has to be provided in the URL (as its last segment). The new
 The JSON format of the request is described in the TypeScript interface ``changeResourceLabelRequest`` in module ``createResourceFormats``.
 The response is described in the TypeScript interface ``changeResourceLabelResponse`` in module ``createResourceFormats``.
 
-*********************************************
-Adding Multiple Resources in a Single Request
-*********************************************
+Bulk Import
+-----------
 
-Multiple resources can be created in a single request. This is especially
-useful if the resources have links to one another. The entire request will be
-checked for consistency as a whole.
+If you have a large amount of data to import into Knora, it can be more convenient to use
+the bulk import feature than to create resources one by one. In a bulk import operation,
+you submit an XML document to Knora, describing multiple resources to be created.
+This is especially useful if the resources to be created have links to one another.
+Knora checks the entire request for consistency as as a whole, and performs the update
+in a single database transaction.
 
-The resources to be created must be described in an XML file. The XML file
-containing the resource descriptions can be imported directly to Knora by a
-POST request. The request has to be sent to the Knora server using the
-``resources/xml`` path segment.
+The procedure for using this feature is as follows:
 
-::
+1. Make a request to the Knora API server to get XML schemas describing the XML to be provided
+   for the import.
+2. If the project's data includes digital representations to be stored in Sipi, upload those files to
```

Comment:
In any case, all the files have to be uploaded to Sipi before any data is added to the triplestore. So the import really has two distinct parts: image upload and triplestore update. The image upload part is going to be a long, error-prone process in which Sipi may crash. Then the user will have to submit a bug report to Sipi, the bug will have to be fixed, and the user will have to try again. I think we should keep this process as simple as possible, to reduce the number of things that can go wrong. Therefore, in the image upload part of the import, I think it's better if the user only has to deal with Sipi, rather than Sipi and Knora.

I don't know whether we actually need to generate thumbnails in an import. Maybe we can just use a Sipi route that directly stores the image, without making a thumbnail. From Knora's point of view, all that matters is that the necessary image metadata is provided.

I don't see this as terribly inconvenient, and I'm going to have to do it myself. I'll just write a little script that sends all the images to Sipi and writes the metadata to a CSV file. Then I'll read this CSV file when generating the XML for the import.